src/libraries/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -246,6 +247,29 @@ internal static void Memmove(ref byte dest, ref byte src, nuint len)
                 goto PInvoke;
             }
 
+#if HAS_CUSTOM_BLOCKS
+            if (len >= 128)
+            {
+                // Try to opportunistically align the destination below. The input isn't pinned, so the GC
+                // is free to move the references. We're therefore assuming that reads may still be unaligned.
+                //
+                // dest is more important to align than src because an unaligned store is more expensive
+                // than an unaligned load.
+#if TARGET_AMD64
+                // Align to 64 bytes for AMD64 (important for 512bit SIMD)
```

Comment:
I think we should always align to 64 bytes here to keep it simple and match Block64 used below.

Reply:
Yeah, also wanted to do that, it probably will be a sub-ns less efficient for arm64 since it doesn't benefit from alignment higher than 16 bytes but it's not worth the complexity

Reply:
Does it maybe make sense to bump the minimum size for this to 256 in this case?

Reply:
Will measure and adjust